my-webxr-app/src/components/LocalCsvReader.tsx

```diff
@@ -9,99 +9,113 @@ import { openDB } from 'idb';
- * @throws {Error} Will throw an error if the database does not exist, or if the store does not exist within the
- * database.
+ * @throws {Error} Will throw an error if the database does not exist, or if the store does not
+ * exist within the database.
  */
 export const validateDbAndStore = async (dbName: string, storeName: string) => {
-    const db = await openDB(dbName, 1);
-    const dbExists = db !== undefined;
-    if (!dbExists) {
-        throw new Error(`Database "${dbName}" does not exist`);
-    }
+  const db = await openDB(dbName, 1);
+  const dbExists = db !== undefined;
+  if (!dbExists) {
+    throw new Error(`Database "${dbName}" does not exist`);
+  }
 
-    const storeExists = db.objectStoreNames.contains(storeName);
-    if (!storeExists) {
-        throw new Error(`Store "${storeName}" does not exist in database "${dbName}"`);
-    }
+  const storeExists = db.objectStoreNames.contains(storeName);
+  if (!storeExists) {
+    throw new Error(`Store "${storeName}" does not exist in database "${dbName}"`);
+  }
 };
 
 /**
  * Handles the parsed data from a CSV file and stores it in a database.
  *
- * @param {any} results - The results object returned by Papa.parse. The object data should be an array.
+ * @param {any} results - The results object returned by Papa.parse. The object data should be an
+ * array.
  * @param {string} dbName - The name of the database where the data should be stored.
- * @param {string} storeName - The name of the store within the database where the data should be stored.
+ * @param {string} storeName - The name of the store within the database where the data should be
+ * stored.
  * @throws {Error} Will throw an error if the 'data' property of the results object is not an array.
  */
-export const handleParsedData = async (results: any, dbName: string, storeName: string) => {
-    if (!Array.isArray(results.data)) {
-        throw new Error('Parsed data must be an array');
-    }
+export const handleParsedData = async (
+  results: Papa.ParseResult<unknown>,
+  dbName: string,
+  storeName: string,
+) => {
+  if (!Array.isArray(results.data)) {
+    throw new Error('Parsed data must be an array');
+  }
 
-    const db = await openDB(dbName, 1);
-    const tx = db.transaction(storeName, 'readwrite');
-    const store = tx.objectStore(storeName);
-    await store.clear(); 
-    for (let i = 0; i < results.data.length; i++) {
-        const item = results.data[i];
-        await store.put(item, i);
-    }
-    await tx.done;
+  const db = await openDB(dbName, 1);
+  const tx = db.transaction(storeName, 'readwrite');
+  const store = tx.objectStore(storeName);
+  await store.clear();
+
+  // Concurrently process each Database push
+  const operations = [];
+  for (let i = 0; i < results.data.length; i += 1) {
+    const item = results.data[i];
+    operations.push(store.put(item, i));
+  }
+  await Promise.all(operations);
+  await tx.done;
```

Comment:
ESLint complained about the repeated `await` with the loop due to poor performance. I refactored them into a `Promise.all` to they all can occur simultaneously. @LongQuanNguyen let me know if you have any questions or thoughts about this.

Reply:
Hmm, idealy we want the transaction to be atomic, that why I used await. If Promise.all() could achieve that then it should be ok. I haven't used promise b4, let me read more about it and get back to you.